base/tuple.jl

```diff
@@ -105,7 +105,8 @@ julia> ntuple(i -> 2*i, 4)
 ```
 """
 function ntuple(f::F, n::Integer) where F
-    t = n <= 0  ? () :
+    (n >= 0) || throw(DomainError())
```

Comment:
`DomainError` changed into `ArgumentError`.
I'd leave the `n` check where it is, because it's much cleaner than burying it inside another routine. Anyway, in performance-sensitive context one should use `Val{N}` alternative.